### PR TITLE
Reimagine Sharing: Add new sharing to Episode detail page

### PR DESCRIPTION
**Pull request by bjtitus**
| 📘 Part of: https://github.com/Automattic/pocket-casts-ios/issues/1910 |
|:---:|

Adds new sharing functionality to Episode Detail page. For now, I left the "Podcast" sharing option in the list because I think it doesn't hurt to have.

| Before | After |
| -- | -- |
| Simulator Screenshot - iPhone 15 Pro - 2024-06-21 at 08 57 14 | Simulator Screenshot - iPhone 15 Pro - 2024-06-21 at 08 58 42 |

## To test

> [!NOTE]
> Enable the `newSharing` feature flag to test this

### New Sharing Modal
* Navigate to an Episode within a podcast from Subscriptions
* Tap the Share button in the upper right
* Ensure share options are shown
* Ensure that the new sharing modal opens when selecting one

### Open downloaded file
* Download an episode
* Tap the Share button
* Ensure that the "Open File in..." option is included in the menu
* Tap the "Open File in..." option and make sure the file is being shared

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/Sharing/SharingModal.swift [hunk 1]
```diff
@@ -60,13 +60,18 @@ enum SharingModal {
 
         let optionPicker = OptionsPicker(title: L10n.share.uppercased(), themeOverride: .dark, colors: colors)
 
-        let actions: [OptionAction] = Option.allCases(episode: episode, podcast: podcast, currentTime: PlaybackManager.shared.currentTime()).map { option in
+        let actions: [OptionAction] = Option.allCases(episode: episode, podcast: podcast, currentTime: episode?.playedUpTo ?? 0).map { option in
```
Comment (SergioEstevao) [inline podcasts/Sharing/SharingModal.swift:63]: Why this change? is the playerUpTo different from currentTime? Just wanted to understand if this is related to this PR.

Comment (bjtitus) [reply podcasts/Sharing/SharingModal.swift:63]: This `SharingModal` is only used with the new sharing design so nothing has changed for our existing sharing.

This just uses the passed in episode's `playedUpTo` time instead of the player's `currentTime` because you can share a link to the current timestamp of an episode without beginning playback. (Podcast > Episode > Share > Current position)

**Review by SergioEstevao — APPROVED** [review]
Looking good! :shipit:

File: podcasts/Sharing/SharingModal.swift (hunk 1 above)
Comment (SergioEstevao) [reply podcasts/Sharing/SharingModal.swift:63]: Thanks for the clarification.
